clients/drcachesim/analysis_tool.h

```diff
@@ -156,6 +156,16 @@ template <typename RecordType> class analysis_tool_tmpl_t {
     {
         return "";
     }
+    /**
+     * Identifies the preferred shard type for this analysis.  If every tool requests
+     * #SHARD_BY_CORE, the framework may decided to use that mode even if the user
+     * left the default thread-sharded mode in place.
+     */
+    virtual shard_type_t
+    preferred_shard_type()
+    {
+        return SHARD_BY_THREAD;
```

Comment:
It should be made clearer that this does NOT guarantee the provided shard type. I know the documentation above only says "may decide" and this function is named "preferred" (and not "asserted_shard_type" or something). But I still think there's room for confusion. Also, see other comment: I don't fully understand the need.

Reply:
Rewrote as:
```
    /**
     * Identifies the preferred shard type for this analysis.  This only applies when
     * the user does not specify a shard type for a run.  In that case, if every tool
     * being run prefers #SHARD_BY_CORE, the framework uses that mode.  If tools
     * disagree then an error is raised.  This is ignored if the user specifies a
     * shard type via one of -core_sharded, -core_serial, -no_core_sharded,
     * -no_core_serial, or -cpu_scheduling.
     */
```